From c4c0a19715eb842b0bc883302558b33d81477a6b Mon Sep 17 00:00:00 2001
From: Danny Kukawka <danny.kukawka@web.de>
Date: Wed, 27 Aug 2008 21:24:33 +0200
Subject: [PATCH 05/11] use udevadm instead of (deprecated) udevinfo

Use udevadm instead of (deprecated) udevinfo. Adapted/fixed version
of a patch contributed by Lucas Hazel <lucas@die.net.au> via HAL ML.
---
 hald/linux/blockdev.c |    2 +-
 hald/linux/coldplug.c |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/hald/linux/blockdev.c b/hald/linux/blockdev.c
index 6042558..9c5ea1f 100644
--- a/hald/linux/blockdev.c
+++ b/hald/linux/blockdev.c
@@ -1750,7 +1750,7 @@ udev_get_device_file_for_sysfs_path (const char *sysfs_path)
         char *ret;
         char *u_stdout;
         int u_exit_status;
-        const char *argv[] = {"/usr/bin/udevinfo", "--root", "--query", "name", "--path", NULL, NULL};
+        const char *argv[] = {"/sbin/udevadm", "info", "--root", "--query", "name", "--path", NULL, NULL};
         GError *g_error;
 
         ret = NULL;
diff --git a/hald/linux/coldplug.c b/hald/linux/coldplug.c
index 18eae78..d77a114 100644
--- a/hald/linux/coldplug.c
+++ b/hald/linux/coldplug.c
@@ -149,8 +149,8 @@ udev_info_to_hotplug_event (const UdevInfo *info)
 static gboolean
 hal_util_init_sysfs_to_udev_map (void)
 {
-	char *udevdb_export_argv[] = { "/usr/bin/udevinfo", "-e", NULL };
-	char *udevroot_argv[] = { "/usr/bin/udevinfo", "-r", NULL };
+	char *udevdb_export_argv[] = { "/sbin/udevadm", "info", "-e", NULL };
+	char *udevroot_argv[] = { "/sbin/udevadm", "info", "-r", NULL };
 	int udevinfo_exitcode;
 	UdevInfo *info = NULL;
 	char *p;
-- 
1.6.0.2

